### PR TITLE
[WebNN] Fallback the node when its output doesn't have shape info

### PR DESCRIPTION
**Pull request by shiyi9801**
WebNN requires that each input and output must have shape info.

### CODE REVIEW

File: onnxruntime/core/providers/webnn/builders/helper.cc [hunk 1]
```diff
@@ -69,26 +69,24 @@ bool IsNodeSupported(const Node& node, const GraphViewer& graph_viewer, const We
   }
 }
 
-bool IsInputSupported(const NodeArg& input, const std::string& parent_name, const logging::Logger& logger) {
-  const auto& input_name = input.Name();
-  const auto* shape_proto = input.Shape();
+bool IsNodeArgSupported(const NodeArg& node_arg, const std::string& parent_name, const logging::Logger& logger) {
```
Comment (Honry) [inline onnxruntime/core/providers/webnn/builders/helper.cc:72]: How about `IsNodeShapeSupported`?

Comment (shiyi9801) [reply onnxruntime/core/providers/webnn/builders/helper.cc:72]: NodeShape sounds like the shape of a node, but it's actually the shape of the operands of a node...

Comment (shiyi9801) [reply onnxruntime/core/providers/webnn/builders/helper.cc:72]: How about `IsInputOutputSupported`?

Comment (shiyi9801) [reply onnxruntime/core/providers/webnn/builders/helper.cc]: Discussed off line, renamed it to `IsTensorShapeSupported`.

**Review by Honry — APPROVED** [review]
👍

**Comment by shiyi9801** [comment]
@fdwr @guschmue PTAL, thanks a lot!

**Review by fdwr — APPROVED** [review]
👍

~~I'm curious what the case was that prompted this? Which model?~~

Ah, a speech to text model.

**Pull request merged**